core/views.py

```diff
@@ -46,7 +46,22 @@ class UnitGroupListView(LoginRequiredMixin, ListView[Unit]):
     template_name = "core/unit_list.html"
 
     def get_queryset(self):
-        queryset = Unit.objects.filter(group__hidden=False)
+        assert isinstance(user := self.request.user, User)
+        student = Student.objects.filter(user=user, semester__active=True).first()
+        if student is None:
+            student = Student.objects.filter(user=user).order_by("-pk").first()
+            active = False
+        else:
+            active = True
+
+        if student:
+            queryset = Unit.objects.exclude(
+                group__hidden=True, group__bonuslevel__level__gt=student.last_level_seen
+            )
+        elif user.is_staff:
```

Comment:
The staff check should come before the student check I think. (Most staff are OTIS alum's!)

Reply:
This is correct, gone and done that